commons-jetty/src/main/scala/com/avsystem/commons/jetty/rpc/JettyRPCFramework.scala

```diff
@@ -16,23 +17,21 @@ import org.eclipse.jetty.server.{Handler, Request}
   * @author MKej
   */
 trait JettyRPCFramework extends StandardRPCFramework {
-  protected def valueToJson(value: RawValue): String
-  protected def jsonToValue(json: String): RawValue
-
-  protected def argsToJson(args: List[List[RawValue]]): String
-  protected def jsonToArgs(json: String): List[List[RawValue]]
+  type RawValue = String
+  type Reader[T] = GenCodec[T]
+  type Writer[T] = GenCodec[T]
 
   class RPCClient(httpClient: HttpClient, urlPrefix: String)(implicit ec: ExecutionContext) {
     private class RawRPCImpl(pathPrefix: String) extends RawRPC {
       def fire(rpcName: String, argLists: List[List[RawValue]]): Unit =
-        put(pathPrefix + rpcName, argsToJson(argLists))
+        put(pathPrefix + rpcName, write(argLists))
```

Comment:
I would say this is not really what we want. We're going to encode a String containing JSON to a JSON String. Yo dawg I herd you like escaping.